PowerShellTools/Intellisense/IntelliSenseManager.cs

```diff
@@ -288,6 +261,51 @@ private void StartIntelliSense(int lineStartPosition, int caretPosition, string
             _intellisenseRunning = false;
         }
 
+        private CommandCompletion GetCommandCompletionList(int caretPosition)
+        {
+            Ast ast;
+            Token[] tokens;
+            IScriptPosition cursorPosition;
+            GetCommandCompletionParameters(caretPosition, out ast, out tokens, out cursorPosition);
+            if (ast == null)
+            {
+                return null;
+            }
+
+            var ps = PowerShell.Create();
+            ps.Runspace = PowerShellToolsPackage.Debugger.Runspace;
+
+            return CommandCompletion.CompleteInput(ast, tokens, cursorPosition, null, ps);
+        }
+
+        private void GetCommandCompletionParameters(int caretPosition, out Ast ast, out Token[] tokens, out IScriptPosition cursorPosition)
+        {
+            ITrackingSpan trackingSpan;
+            _textView.TextBuffer.Properties.TryGetProperty(BufferProperties.Ast, out ast);
+            _textView.TextBuffer.Properties.TryGetProperty(BufferProperties.Tokens, out tokens);
+            _textView.TextBuffer.Properties.TryGetProperty(BufferProperties.SpanTokenized, out trackingSpan);
+
+            // No ast or tokens available on the buffer. Re-tokenize.
+            if (ast == null || tokens == null)
+            {
+                var trackingSpan2 = _textView.TextBuffer.CurrentSnapshot.CreateTrackingSpan(0, _textView.TextBuffer.CurrentSnapshot.Length, SpanTrackingMode.EdgeInclusive);
+                var text = trackingSpan2.GetText(_textView.TextBuffer.CurrentSnapshot);
+                ParseError[] array;
+                ast = CommandCompletionHelper.Tokenize(text, out tokens, out array);
+            }
+            if (ast != null)
+            {
+                //HACK: Clone with a new offset using private method... 

```

Comment:
Similar but not the same. I put the methods needed by both in-proc and out-proc in CommandCompletionHelper.